types/block_test.go

```diff
@@ -367,6 +368,98 @@ func TestCommitToVoteSet(t *testing.T) {
 	}
 }
 
+func TestCommitToVoteSetWithVotesForAnotherBlock(t *testing.T) {
+	lastID := makeBlockIDRandom()
+	height := int64(3)
+	round := 1
+	numValidators := 10
+
+	voteSet, _, vals := randVoteSet(height-1, 1, PrecommitType, numValidators, 1)
+
+	// > 2/3 sign for this block
+	numBlockValidators := numValidators*2/3 + 1
+	for i := 0; i < numBlockValidators; i++ {
+		addr := vals[i].GetPubKey().Address()
+		vote := &Vote{
+			ValidatorAddress: addr,
+			ValidatorIndex:   i,
+			Height:           height - 1,
+			Round:            round,
+			Type:             PrecommitType,
+			BlockID:          lastID,
+			Timestamp:        tmtime.Now(),
+		}
+
+		_, err := signAddVote(vals[i], vote, voteSet)
+		assert.NoError(t, err)
+	}
+
+	// 1/3 vote for a different block
```

Comment:
```suggestion
	// 1/3 vote for different blocks
```

every call to `makeBlockIDRandom` produces different value imo

Reply:
of course! good catch!